src/border_agent/border_agent.cpp

```diff
@@ -77,6 +77,8 @@ static const char    kBorderAgentServiceType[]      = "_meshcop._udp";   ///< Bo
 static const char    kBorderAgentEpskcServiceType[] = "_meshcop-e._udp"; ///< Border agent ePSKc service
 static constexpr int kBorderAgentServiceDummyPort   = 49152;
 
+static constexpr uint16_t kThreadVersion14 = 5; ///< Thread Version 1.4
```

Comment:
Maybe you can just use [`OT_THREAD_VERSION_1_4`](https://github.com/openthread/openthread/blob/df757ba0cf3405e5d36538f3c774500a08ff366e/src/core/openthread-core-config.h#L46)?

Reply:
This definition is in core and could not be used directly? Open an issue to track refactor of thread version in OTBR b/358571361

Reply:
Sorry I didn't pay attention to where the macro was defined. 

IMHO we should expose the `OT_THREAD_VERSION_*` macros as part of Openthread API.  Currently [otThreadGetVersion](https://github.com/openthread/openthread/blob/df757ba0cf3405e5d36538f3c774500a08ff366e/include/openthread/thread.h#L252) doesn't tell what the returned integer means.

@abtink @jwhui what do you think?

Reply:
@superwhd Just FYI, there is a dbus API [GetThreadVersion](https://github.com/openthread/ot-br-posix/blob/1dc6c81d2721f9dc3ccbdf343fb5d0ecc391d574/src/ncp/rcp_host.cpp#L372) can tell the version mapping (slightly differences of version1.3.1); 

Temperately add version definition in BA in this PR and try to refactor OT version in OTBR, tracking here: b/358571361. Appreciate to leave comments for the refactor.

Reply:
> IMHO we should expose the OT_THREAD_VERSION_* macros as part of Openthread API. Currently [otThreadGetVersion](https://github.com/openthread/openthread/blob/df757ba0cf3405e5d36538f3c774500a08ff366e/include/openthread/thread.h#L252) doesn't tell what the returned integer means.

I 100% agree. I was expecting `OT_THREAD_VERSION_*` to be defined in `thread.h` header, but for some reason they are  defined in `openthread-core-config.h` (which is not a public header). I will submit a PR on this.